File: 05.11.25/recurtion.cpp
#include <iostream>
#include <string>
using namespace std;

string reverseString(string s) {
    // Базовый случай
    if (s.length() <= 1) {
        return s;
    }
    // Рекурсивный случай
    return s.back() + reverseString(s.substr(0, s.length() - 1));
}

int main() {
    string testStr = "helloWorld!";
    cout << "Оригинал: " << testStr << endl;
    cout << "Перевёрнутая: " << reverseString(testStr) << endl;
    return 0;
}

//Вывод:
//Оригинал: helloWorld!
//Перевёрнутая: !dlroWolleh

File: 05.11.25/recurtion.py
def reverse_string(s):
    # Базовый случай: если строка пустая или состоит из одного символа
    if len(s) <= 1:
        return s
    # Рекурсивный случай: последний символ + рекурсивный вызов для остальной части
    return s[-1] + reverse_string(s[:-1])

# Тестирование
test_str = "hello"
print(f"Оригинал: {test_str}")
print(f"Перевёрнутая: {reverse_string(test_str)}")

#Вывод
#Оригинал: hello
#Перевёрнутая: olleh
